include/byte_order_generic_x86_64.h

```diff
@@ -17,65 +17,88 @@
   Optimized function-like macros for the x86 architecture (_WIN32 included).
 */
 
-#define sint2korr(A)	(int16) (*((int16 *) (A)))
-#define sint3korr(A)	((int32) ((((uchar) (A)[2]) & 128) ? \
-				  (((uint32) 255L << 24) | \
-				   (((uint32) (uchar) (A)[2]) << 16) |\
-				   (((uint32) (uchar) (A)[1]) << 8) | \
-				   ((uint32) (uchar) (A)[0])) : \
-				  (((uint32) (uchar) (A)[2]) << 16) |\
-				  (((uint32) (uchar) (A)[1]) << 8) | \
-				  ((uint32) (uchar) (A)[0])))
-#define sint4korr(A)	(int32)  (*((int32 *) (A)))
-#define uint2korr(A)	(uint16) (*((uint16 *) (A)))
-#define uint3korr(A)	(uint32) (((uint32) ((uchar) (A)[0])) +\
-				  (((uint32) ((uchar) (A)[1])) << 8) +\
-				  (((uint32) ((uchar) (A)[2])) << 16))
-#define uint4korr(A)	(uint32) (*((uint32 *) (A)))
-
-
-static inline ulonglong uint5korr(const void *p)
+#include <string.h>
+
+inline static int16 sint2korr(const void *p)
+{
+  int16 result= 0;
```

Comment:
Why `= 0`?

Reply:
I like to avoid uninitialized variables. Compiler will optimize such a simple dead store.

Reply:
Why uninitialised? Don't you initialise it 1 line below? Besides I normally do want uninitialised access warning (not in this particular case though) from the compiler, which would be suppressed by this initialisation.

Reply:
I mean initialization at the same statement where declaration happens. It's just my habit. I may remove this if you like.

Reply:
It means you rely on compiler to optimise it away. Also, wouldn't it hide otherwise uninitialised access from tools like ASAN or from `-Wuninitialized`?

Reply:
Yes, it will hide it. It's a programmer's job to ensure that variable initialized with a correct value.